packages/frontend/hooks/src/use-block-suite-editor.ts

```diff
@@ -0,0 +1,30 @@
+import type { AffineEditorContainer } from '@blocksuite/presets';
+import { atom, useAtom } from 'jotai';
+import { useCallback, useMemo } from 'react';
+
+export const editorContainerAtom = atom<{
+  [editorId: string]: AffineEditorContainer | null;
+}>({});
+
+export function useBlocksuiteEditor(pageId: string) {
+  const [editors, setEditors] = useAtom(editorContainerAtom);
+
+  const editor = editors[pageId];
+
+  const setEditorContainer = useCallback(
+    (editorContainer: AffineEditorContainer) => {
+      setEditors(prevEditors => ({
+        ...prevEditors,
+        [pageId]: editorContainer,
+      }));
+    },
+    [setEditors, pageId]
+  );
+
+  return useMemo(() => {
+    return {
+      editor,
+      setEditorContainer,
```

Comment:
the name of the return value may use `useState` pattern that returns a 2-tuple.
also the name of the first param should be `editorContainer`